app/code/Magento/Quote/Model/QuoteRepository/Plugin/AccessChangeQuoteControl.php

```diff
@@ -53,7 +53,7 @@ public function beforeSave(CartRepositoryInterface $subject, CartInterface $quot
      * @param Quote $quote
      * @return bool
      */
-    private function isAllowed(Quote $quote)
+    public function isAllowed(Quote $quote)
```

Comment:
There is no real use for making any additional methods in a plugin `public` (apart from the plugin itself):
* Technical Guidelines do not recommend injecting a plugin into any class, it is not supposed to work that way.
* It is not a really good idea to create a plugin for another plugin.

In this case I suggest to introduce a separate service which would provide this logic, thus, it will be available for any third party code and may be used in this plugin at the same time.